File: 3152. Special Array II.go
func isArraySpecial(nums []int, queries [][]int) []bool {
	prefix := make([]int, len(nums))
	nums[0] = nums[0] % 2
	for i := 1; i < len(nums); i++ {
		nums[i] = nums[i] % 2
		if nums[i] != nums[i-1] {
			prefix[i] = 0
		} else {
			prefix[i] = 1
		}
		prefix[i] += prefix[i-1]
	}
	answer := make([]bool, len(queries))
	for i := 0; i < len(answer); i++ {
		left, right := queries[i][0], queries[i][1]
		if left == 0 {
			answer[i] = prefix[right] == 0
		} else {
			answer[i] = prefix[right] == prefix[left]
		}
	}
	return answer
}

File: 3152. Special Array II.java
class Solution {
    public boolean[] isArraySpecial(int[] nums, int[][] queries) {
        int[] prefix = new int[nums.length];
        nums[0] = nums[0] % 2;
        for (int i = 1; i < nums.length; i++) {
            nums[i] = nums[i] % 2;
            prefix[i] = nums[i] != nums[i-1] ? 0 : 1;
            prefix[i] += prefix[i-1];
        }
        boolean[] answer = new boolean[queries.length];
        for (int i = 0; i < answer.length; i++) {
            int left = queries[i][0], right = queries[i][1];
            answer[i] = left == 0 ? prefix[right] == 0 : prefix[right] == prefix[left];
        }
        return answer;
    }
}